app/controllers/twitter.js

```diff
@@ -1,23 +1,210 @@
+}
+
+important fields: 
+	create_at	: UTC time when this Tweet was created.
+	coordinates	: he longitude and latitude of the Tweet’s location.
+	text		: The actual UTF-8 text of the status update.T.
 
-let Writable = require('stream').Writable,
+*/
+
+let fs = require('fs'),
+    config = require('../../config'),
+	Writable = require('stream').Writable,
     TwitterStream = require('twitter-stream-api'),
-    config = require('../../config');
+	pokemonTweet = require(__appbase + 'models/pokemonAppearances.js'),
+    pokemonListPath = require(__appbase + '../resources/json/pokemonIdAndName.json');
 
 const Twitter = new TwitterStream(config.twitter);
 
+/*
+	keywords used to search for in the tweets
+	,  : represents OR
+	' ': space represents AND 
+*/
+const tweetSearchKeywords = 'pokemon catch,pokemon saw,pokemon attack,pokemon find,pokemon caught,pokemon attacked,pokemon found,pokemon appeared,#foundPokemon,#caughtPokemon,#pokemongo,a wild appeared until #pokemongo';
+
+// array of names of pokemons
+var pokemonNameList = [];
+
+// process the tweet object
 let Output = Writable({objectMode: true});
 Output._write = function (obj, enc, next) {
-    // the tweet json
-    console.log(JSON.stringify(obj));
-    // the tweet id and text
-    console.log(obj.id, obj.text);
+	checkIfTweetHasPokemonName(obj);
     next();
 };
 
+/*
+	check if tweet object has pokemon term
+ */
+function checkIfTweetHasPokemonName(tweetObj) {
+	// check if the tweet text contains the important keywords
+	if (tweetObj.text.toLowerCase().indexOf('caught') != -1 || 
+		tweetObj.text.toLowerCase().indexOf('saw') != -1 || 
+		tweetObj.text.toLowerCase().indexOf('attacked') != -1 ||
+		tweetObj.text.toLowerCase().indexOf('found') != -1  ||
+		tweetObj.text.toLowerCase().indexOf('appeared') != -1 ) {
+
+		// for each pokemon name check if it is inside the tweet text
+		for(var i = 0; i < pokemonNameList.length; i++) {
+			if (tweetObj.text.toLowerCase().indexOf(pokemonNameList[i]) != -1){
+				//get pokemon names array
+				console.log(tweetObj.text.toLowerCase());
+				console.log(pokemonNameList[i] + " was found");
+				addPokemonAppearanceToDB(tweetObj, pokemonNameList[i]);
+			}
+		}
+	}
+}
+
+/*
+	add pokemon appearance to DB 
+	properties: tweetId (unique), pokemonName, foundAt (latitude,longitude), appearedOn: (DateString UTC)
+ */
+
+function addPokemonAppearanceToDB(tweetObj, pokemonName) {
+	//default pokemon latitude and longitude set to -1, new values only if tweet contains coordinates
+	let pokemonFoundLongitude = -1, pokemonFoundLatitude = -1;
+	//get the location of the tweet
+	if (tweetObj.coordinates){
+		pokemonFoundLongitude = tweetObj.coordinates.coordinates[0]
+		pokemonFoundLatitude = tweetObj.coordinates.coordinates[1]
+	}
+
+	pokemonTweet.create({tweetId: tweetObj.id, pokemonName: pokemonName, foundAt: {latitude: pokemonFoundLatitude, longitude: pokemonFoundLongitude}, appearedOn: tweetObj.created_at}, function (err, post) {
```

Comment:
twitter recommends to use id_str instead of id (https://dev.twitter.com/overview/api/twitter-ids-json-and-snowflake)